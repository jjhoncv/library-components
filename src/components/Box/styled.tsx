import styled from 'styled-components';

export const Box = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 34px 34px 0;
  width: 100%;
  position: relative;
  z-index: 1;
`;
