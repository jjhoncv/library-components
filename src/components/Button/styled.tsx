import styled, { keyframes } from 'styled-components';
import { COLORS, FONTS } from './../../ui'

interface Props {
  fullWidth?: boolean;
  disabled?: boolean;
  loading?: boolean;
  primary?: boolean;
}

const animationLoading = keyframes`
  to {transform: rotate(360deg);}
`;

export const ButtonLoading = styled.div`
  display: inline-block;
  box-sizing: border-box;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 2px solid ${COLORS.BLACK};
  border-top: 2px solid ${COLORS.LIGHT_GRAY};
  animation: ${animationLoading} .6s linear infinite;

  ${(props: Props) => props.primary && `
    border: 2px solid ${COLORS.WHITE};
    border-top: 2px solid ${COLORS.BLACK};
  `};
`;

export const ButtonStyled = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${(props: Props) => props.fullWidth ? '100%' : 'auto'};
  background-color: ${COLORS.LIGHT_GRAY};
  height: 44px;
  padding: 12px 25px;
  border: none;
  border: 1px solid ${COLORS.OPAQUE_GRAY};
  border-radius: 4px;
  color: ${COLORS.BLACK};
  font-size: 14px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  cursor: pointer;

  &:hover {
    background-color: ${COLORS.HOVER_GRAY};
  }

  ${(props: Props) => props.disabled && `
    pointer-events: none;
    user-select: none;
    opacity: 0.3;
  `};

  ${(props: Props) => props.loading && `
    pointer-events: none;
    user-select: none;
  `};

  ${(props: Props) => props.primary && `
    background:${COLORS.BLACK};
    color: ${COLORS.WHITE};

    &:hover {
      background-color: ${COLORS.HOVER_BLACK};
    }
  `};
`;
