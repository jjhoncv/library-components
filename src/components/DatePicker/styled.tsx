import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

interface Props {
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
}

interface WeekDateProps {
  disabled?: boolean;
  selected?: boolean;
}

export const DatePickerWrapper = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  font-family: ${FONTS.ROBOTO_REGULAR};

  ${(props: Props) => props.disabled && `
    pointer-events: none;

    ${DatePickerBox} {
      opacity: 0.3;
    }
  `}
`;

export const DatePickerIcon = styled.div`
  display: flex;
  flex-wrap: wrap;
  position: absolute;
  right: 8px;
  top: 8px;
  cursor: pointer;
  z-index: 1;

  ${(props: Props) => props.error && `
    svg {
      color: black;
    }
  `}
`;

export const DatePickerBox = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
  position: relative;
  z-index: 0;
  color: ${COLORS.BLACK};
  font-size: 14px;
`;

export const DatePickerInput = styled.input`
  background-color: transparent;
  width: 100%;
  height: 40px;
  padding: 12px 30px 12px 12px;
  border: 2px solid ${COLORS.BORDER_GRAY};
  box-sizing: border-box;
  border-radius: 4px;
  cursor: pointer;
  z-index: 2;
  &::placeholder {
    font-family: ${FONTS.ROBOTO_REGULAR};
  }
  &:focus {
    outline: none;
    border: 2px solid ${COLORS.AQUA_BLUE};
  }

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  &:focus ~ ${DatePickerIcon} {
    svg {
      color: black;
    }
  }
`;

export const DatePickerCalendarWrapper = styled.div`
  position: absolute;
  margin-top: 40px;
  z-index: 9;
  width: 240px;
  max-height: 300px;
  background-color: ${COLORS.WHITE};
  color: ${COLORS.BLACK};
  border: 1px solid ${COLORS.BORDER_GRAY};
  box-shadow: 0px 0px 8px ${COLORS.HOVER_GRAY};
  border-radius: 4px;
  padding: 5px;
`;

export const DatePickerMonth = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: bold;
`;

export const DatePickerPrev = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerNext = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerTitle = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 5 1 0;
  justify-content: center;
`;

export const DatePickerWeekDays = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
`;

export const DatePickerWeekDay = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  font-weight: bold;
`;

export const DatePickerDates = styled.div``;

export const DatePickerWeek = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export const DatePickerWeekDate = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  ${(props: WeekDateProps) => props.disabled && `
    color: ${COLORS.TEXT_GRAY} !important;
    cursor: not-allowed;

    &: hover {
      background: ${COLORS.WHITE} !important;
      color: ${COLORS.TEXT_GRAY} !important;
    }
  `}

  ${(props: WeekDateProps) => props.selected && `
    background-color: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  `}

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;


