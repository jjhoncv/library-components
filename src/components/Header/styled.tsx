import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

export const HeaderWrapper = styled.div`
  width: 100%;
`;
export const HeaderTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_BOLD};
  font-size: 24px;
  line-height: 30px;
  padding-bottom: 7px;
  border-bottom: 1px black dashed;
  color: ${COLORS.TEXT_GRAY};
`;
export const HeaderSubTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_REGULAR};
  font-size: 18px;
  line-height: 23px;
  color: ${COLORS.BLACK};
  padding-top: 9px;
`;

export const HeaderTitleImage = styled(HeaderTitle)`
  display: flex;
  ${({ pos }) => (pos === "start") && `
    justify-content: flex-end;
    flex-direction: row-reverse;
    align-items: flex-end;
  `}

  ${({ pos }) => (pos === "end") && `
    align-items: flex-end;
    justify-content: space-between;
  `}
`;
