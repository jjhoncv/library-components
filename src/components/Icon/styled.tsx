import styled from 'styled-components';
import { COLORS } from './../../ui';

interface Props {
  color?: string,
  size?: string,
  colorHover?: string
}

export const IconStyled = styled.div`
  display: inline-flex;
  flex-wrap: wrap;

  svg {
    color: ${(props: Props) => props.color || `${COLORS.TEXT_GRAY}`}
    ${(props: Props) => `
      width: ${props.size || 'auto'};
      height: ${props.size || 'auto'};
    `}
  }

  &:hover {
    svg {
      color: ${(props: Props) => props.colorHover && `${props.colorHover}`};
    }
  }
`;
