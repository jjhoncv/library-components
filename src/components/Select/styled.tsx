import styled from 'styled-components';
import { COLORS, FONTS } from '../../ui';

interface Props {
  disabled?: boolean;
  error?: boolean;
  width?: string;
  fullWidth?: boolean;
}

export const SelectIcon = styled.div`
  display: flex;
  flex-wrap: wrap;
  position: absolute;
  right: 8px;
  top: 8px;
  cursor: pointer;
  z-index: 1;
`;


export const SelectStyled = styled.select`
  width: 100%;
  height: 40px;
  border: 2px solid ${COLORS.BORDER_GRAY};
  border-radius: 4px;
  background: transparent;
  box-sizing: border-box;
  padding: 0 12px;
  outline: 0;
  appearance: none;
  cursor: pointer;
  z-index: 2;

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  &:focus {
    border: 2px solid ${COLORS.AQUA_BLUE};
  }

  &:focus ~ ${SelectIcon} {
    svg {
      color: black;
    }
  }
`;


export const SelectWrapper = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  align-items: center;
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  position: relative;
  z-index: 0;
  color: ${COLORS.BLACK};
  font-family: ${FONTS.ROBOTO_REGULAR};
  font-size: 14px;

  ${(props: Props) => props.disabled && `
    pointer-events: none;

    ${SelectStyled} {
      opacity: 0.3;
    }

    ${SelectIcon} {
      opacity: 0.3;
    }
  `}
`;

