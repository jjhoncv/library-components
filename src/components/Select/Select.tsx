import * as React from 'react';
import * as PropTypes from 'prop-types';
import { useState } from 'react';
import { SelectWrapper, SelectStyled, SelectIcon } from './styled';
import { Icon } from './../Icon';

interface OptionProps {
  value?: number;
  label?: string;
}

interface Props {
  placeholder?: string;
  disabled?: boolean;
  error?: boolean;
  width?: string;
  fullWidth?: boolean;
  selected?: string;
  options: OptionProps[]
}

export const Select: React.FC<Props> = ({
  placeholder,
  disabled,
  error,
  width,
  fullWidth,
  selected,
  options
}) => {

  const [ value, setValue ] = useState(selected);

  const handleChange = (e) => {
    setValue(e.target.value);
  }

  return (
    <SelectWrapper
      disabled={disabled}
      fullWidth={fullWidth}
      width={width}>
      <SelectStyled
        error={error}
        value={value}
        onChange={handleChange}
      >
        {placeholder && <option value="">{placeholder}</option>}
        {options.map((item, index) => (
          <option key={index} value={item.value}>
            { item.label }
          </option>
        ))}
      </SelectStyled>
      <SelectIcon>
        <Icon name="chevron-down" color="#B6B6B6"/>
      </SelectIcon>
    </SelectWrapper>
  );
}

Select.propTypes = {
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  error: PropTypes.bool,
  width: PropTypes.string,
  fullWidth: PropTypes.bool,
  selected: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.number,
    label: PropTypes.string,
  })).isRequired
}

Select.defaultProps = {
  width: '200px'
}
