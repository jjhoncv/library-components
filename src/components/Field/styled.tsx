import styled from 'styled-components';
import { SelectWrapper } from '../Select/styled';
import { DatePickerWrapper } from '../DatePicker/styled';
import { TimePickerWrapper } from '../TimePicker/styled';
import { TextAreaStyled } from '../TextArea/styled';

import { COLORS, FONTS } from '../../ui';

interface GroupProps {
  keepProps?: boolean;
}

interface Props {
  errorMessage?: string;
  portionWidth?: number;
}

export const FieldStyled = styled.div`
  font-family: ${FONTS.ROBOTO_REGULAR};
  width: ${props => `${props.portionWidth * 100}%`};
  flex: 1 1 100%;


  input {
    width: 100%;
  }
  ${SelectWrapper} {
    width: 100%;
  }
  ${DatePickerWrapper} {
    width: 100%;
  }
  ${TimePickerWrapper} {
    width: 100%;
  }
  ${TextAreaStyled} {
    width: 100%;
  }
  margin-bottom: 14px;

  @media(min-width: 480px) {
    ${(props: Props) => props.portionWidth && `
      flex: 1 1 calc(${props.portionWidth * 100}% - 20px);
    `}
  }
`;

export const FieldGroup = styled.div`
  display: flex;
  flex-wrap: wrap;

  ${(props: GroupProps) => props.keepProps && `
    ${FieldStyled} {
      margin-right: 20px;
      flex: 1 1 calc(50% - 20px);
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  `}

  @media(min-width: 480px) {
    ${FieldStyled} {
      margin-right: 20px;
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  }
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 2px;
  min-height: 19px;
  color: ${COLORS.BLACK};
  font-size: 12px;

  @media(min-width: 480px) {
    font-size: 14px;
  }
`;

export const BoxMessage = styled.span`
  display: block;
  font-size: 12px;
  line-height: 15px;
  margin-top: 4px;
  ${(props: Props) => props.errorMessage && `
    color: ${COLORS.ERROR_RED};
  `}
  height: 15px;
`;
