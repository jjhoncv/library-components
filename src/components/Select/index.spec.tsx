import * as React from 'react';
import { shallow } from 'enzyme';
import { Select } from './Select';
import { SelectStyled } from './styled';

const options = [
  {
    label: 'Soles',
    value: 0
  },
  {
    label: 'DÃ³lares',
    value: 1
  }
];

describe('Select Component', () => {
  describe('Select => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Select options={options}/>
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Select Builder', () => {
    it('Select => Render prop disabled', () => {
      const wrapper = shallow(
        <Select options={options} disabled/>
      );
      expect(wrapper.prop('disabled')).toBe(true);
    });
    it('Select => Render prop error', () => {
      const wrapper = shallow(
        <Select options={options} error/>
      );
      expect(wrapper.find(SelectStyled).prop('error')).toBe(true);
    });

    it('Select => Render label of the first option', () => {
      const wrapper = shallow(
        <Select options={options}/>
      );
      expect(wrapper.find(SelectStyled).childAt(0).prop('children')).toBe('Soles');
    });

    it('Select => Render placeholder', () => {
      const wrapper = shallow(
        <Select options={options} placeholder="Seleccione monto"/>
      );
      expect(wrapper.find(SelectStyled).childAt(0).prop('children')).toBe('Seleccione monto');
    });
  });
});
