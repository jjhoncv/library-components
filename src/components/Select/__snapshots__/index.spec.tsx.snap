// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Select => render snapshot Render correctly 1`] = `"<div width=\\"200px\\" class=\\"sc-htpNat eARpma\\"><select class=\\"sc-bwzfXH hzUOpi\\"><option value=\\"0\\">Soles</option><option value=\\"1\\">DÃ³lares</option></select><div class=\\"sc-bdVaJa hGBZVw\\"><div color=\\"#B6B6B6\\" class=\\"sc-bxivhb iJWmdf\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><polyline points=\\"6 9 12 15 18 9\\"></polyline></svg></div></div></div>"`;
